src/components/shell/shell.js

```diff
@@ -246,14 +253,33 @@ class Shell extends Component {
    * @memberof Shell
    */
   _onWindowResize(): void {
+    this._updatePlayerClientRect();
+    if (document.body) {
+      this.props.updateDocumentWidth(document.body.clientWidth);
+    }
+  }
+
+  /**
+   * document scroll handler
+   *
+   * @returns {void}
+   * @memberof Shell
+   */
+  _onDocumentScroll(): void {
```

Comment:
If we don't have any more logic in it, no need this function, define just `_updatePlayerClientRect` as a callback for the document scroll event.